ads: unify ads protocol handling across type urls

ramaraochavali opened the pull request:
This PR unifies the ads protocol handling logic across type urls using `WatchedResources` and also removes inconsistencies.
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

istio-testing commented:
Skipping CI for Draft Pull Request.
If you want CI signal for your change, please convert it to an actual PR.
You can still manually trigger a test run with `/test all`

ramaraochavali commented:
/test all

istio-testing commented:
@ramaraochavali: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | 9516aac53c22441c533a3afd67aac54e5e05fd1b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/lint_istio/15228) | `/test lint_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@ramaraochavali: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | 9516aac53c22441c533a3afd67aac54e5e05fd1b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/lint_istio/15228) | `/test lint_istio`
gencheck_istio | 9516aac53c22441c533a3afd67aac54e5e05fd1b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/gencheck_istio/13819) | `/test gencheck_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@ramaraochavali: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | 9516aac53c22441c533a3afd67aac54e5e05fd1b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/lint_istio/15228) | `/test lint_istio`
gencheck_istio | 9516aac53c22441c533a3afd67aac54e5e05fd1b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/gencheck_istio/13819) | `/test gencheck_istio`
unit-tests_istio | 9516aac53c22441c533a3afd67aac54e5e05fd1b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/unit-tests_istio/15790) | `/test unit-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@ramaraochavali: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | 9516aac53c22441c533a3afd67aac54e5e05fd1b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/lint_istio/15228) | `/test lint_istio`
gencheck_istio | 9516aac53c22441c533a3afd67aac54e5e05fd1b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/gencheck_istio/13819) | `/test gencheck_istio`
unit-tests_istio | 9516aac53c22441c533a3afd67aac54e5e05fd1b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/unit-tests_istio/15790) | `/test unit-tests_istio`
integ-pilot-k8s-tests_istio | 9516aac53c22441c533a3afd67aac54e5e05fd1b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-pilot-k8s-tests_istio/15346) | `/test integ-pilot-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

ramaraochavali commented:
/test all

istio-testing commented:
@ramaraochavali: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-pilot-k8s-tests_istio | 9516aac53c22441c533a3afd67aac54e5e05fd1b | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-pilot-k8s-tests_istio/15346) | `/test integ-pilot-k8s-tests_istio`
unit-tests_istio | 6d6a509de94063268ccc918ad27095ae9ba07720 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/unit-tests_istio/15794) | `/test unit-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

howardjohn commented:
But I am happy to see this!

ramaraochavali commented:
Sure. No problem

istio-testing commented:
@ramaraochavali: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | 6d6a509de94063268ccc918ad27095ae9ba07720 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/unit-tests_istio/15794) | `/test unit-tests_istio`
integ-pilot-k8s-tests_istio | 6d6a509de94063268ccc918ad27095ae9ba07720 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-pilot-k8s-tests_istio/15350) | `/test integ-pilot-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

howardjohn commented:
By the way in my I tried to refactor and modify the logic - I think this is
just refactorings (?)

On Fri, Jul 3, 2020, 7:15 AM Istio Automation <notifications@github.com>
wrote:

> @ramaraochavali <https://github.com/ramaraochavali>: The following tests
> *failed*, say /retest to rerun all failed tests:
> Test name Commit Details Rerun command
> unit-tests_istio 6d6a509
> <https://github.com/istio/istio/commit/6d6a509de94063268ccc918ad27095ae9ba07720>
> link
> <https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/unit-tests_istio/15794> /test
> unit-tests_istio
> integ-pilot-k8s-tests_istio 6d6a509
> <https://github.com/istio/istio/commit/6d6a509de94063268ccc918ad27095ae9ba07720>
> link
> <https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-pilot-k8s-tests_istio/15350> /test
> integ-pilot-k8s-tests_istio
>
> Instructions for interacting with me using PR comments are available here
> <https://git.k8s.io/community/contributors/guide/pull-requests.md>. If
> you have questions or suggestions related to my behavior, please file an
> issue against the kubernetes/test-infra
> <https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
> repository. I understand the commands that are listed here
> <https://go.k8s.io/bot-commands>.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/istio/istio/pull/25192#issuecomment-653568535>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAEYGXKIBRZBKYDZGAKZLKTRZXRX5ANCNFSM4OPWLXIQ>
> .
>

ramaraochavali commented:
It also fixes inconsistencies of stale nonce handling across resource types. Rds handles it  and Eds does not handle stale nonce at all. I am and trying to fix the inconsistency and handle the protocol uniformly across all type urls.

ramaraochavali commented:
PTAL when you get chance- if there is an overlap with the other PR, we can wait till that is merged and see if this is needed or we can decide what is the best way to move forward based on what you think. My main intention was to refactor the protocol handling to a separate function so that the inconsistencies can be avoided

howardjohn commented:
No let's merge this one first if it works - mine doesn't work. But i think
this may move us to consistently wrong... Our entire xds handling doesn't
really match the expectation of envoy. But we should probably be consistent
then fix things up

On Fri, Jul 3, 2020, 8:03 AM Rama Chavali <notifications@github.com> wrote:

> PTAL when you get chance- if there is an overlap with the other PR, we can
> wait till that is merged and see if this is needed or we can decide what is
> the best way to move forward based on what you think. My main intention was
> to refactor the protocol handling to a separate function so that the
> inconsistencies can be avoided
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/istio/istio/pull/25192#issuecomment-653587287>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAEYGXPTOSAABNIOZJYEGBDRZXXLLANCNFSM4OPWLXIQ>
> .
>

howardjohn commented:
Also yes the test client (adsc) is different than envoy behavior, at the
very least it doesn't include resource names in ack

On Fri, Jul 3, 2020, 8:05 AM John Howard <howardjohn@google.com> wrote:

> No let's merge this one first if it works - mine doesn't work. But i think
> this may move us to consistently wrong... Our entire xds handling doesn't
> really match the expectation of envoy. But we should probably be consistent
> then fix things up
>
> On Fri, Jul 3, 2020, 8:03 AM Rama Chavali <notifications@github.com>
> wrote:
>
>> PTAL when you get chance- if there is an overlap with the other PR, we
>> can wait till that is merged and see if this is needed or we can decide
>> what is the best way to move forward based on what you think. My main
>> intention was to refactor the protocol handling to a separate function so
>> that the inconsistencies can be avoided
>>
>> —
>> You are receiving this because you commented.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/istio/istio/pull/25192#issuecomment-653587287>, or
>> unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAEYGXPTOSAABNIOZJYEGBDRZXXLLANCNFSM4OPWLXIQ>
>> .
>>
>

ramaraochavali commented:
/test all

istio-testing commented:
@ramaraochavali: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | 6d6a509de94063268ccc918ad27095ae9ba07720 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/unit-tests_istio/15794) | `/test unit-tests_istio`
integ-pilot-k8s-tests_istio | 6d6a509de94063268ccc918ad27095ae9ba07720 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-pilot-k8s-tests_istio/15350) | `/test integ-pilot-k8s-tests_istio`
lint_istio | 8ca0ddebba2716ed9dbf0bffff886eb2da4c0153 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/lint_istio/15248) | `/test lint_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

ramaraochavali commented:
/test all

istio-testing commented:
@ramaraochavali: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint_istio | fd8de393434fb792b940aa21bc473020c4264be6 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/lint_istio/15251) | `/test lint_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

ramaraochavali commented:
/test all

ramaraochavali commented:
@howardjohn  PTAL. This is ready for review. All tests are passing (I made few changes to pass both version and nonce so that the protocol works correctly). Removed all inconsistencies across type urls. The new `shouldRespond` handles all the ADS protocol logic. Also moved to `WatchedResources` so that we can build on top of this. Left few TODOs for further work once we are good with this and merged.

istio-testing commented:
@ramaraochavali: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-telemetry-k8s-tests_istio | ccc20242deb415687ba29ac9fe9a367c1a986193 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-telemetry-k8s-tests_istio/15250) | `/test integ-telemetry-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

ramaraochavali commented:
/test integ-telemetry-k8s-tests_istio

ramaraochavali commented on pilot/pkg/xds/ads.go:
```diff
@@ -163,24 +163,20 @@ func (s *DiscoveryServer) receiveThread(con *Connection, reqChannel chan *discov
 	firstReq := true
 	for {
 		req, err := con.stream.Recv()
-		con.mu.RLock()
```
I think this is not needed. We can directly use in if block - we are already accessing directly from con in if block

howardjohn commented on pilot/pkg/xds/ads.go:
```diff
@@ -80,13 +81,10 @@ type Connection struct {
-	RouteVersionInfoSent                  string
-	EndpointNonceSent, EndpointNonceAcked string
+	// Active contains the list of watched resources for the connection, keyed by the TypeUrl.
+	Active map[string]*model.WatchedResource
```
I think this will likely need to be in the proxy not connection. Either that, or we move the TypeUrl part to proxy and keep this in connection. See https://github.com/istio/istio/pull/24877 for context

howardjohn replied:
Actually it probably makes sense to have it in the proxy. That way the configgen isn't doing something weird like looking at nonces.

ramaraochavali replied:
That makes sense. Actually proxy has Active already. I just moved the code to use it. And it will help to unify generators later

howardjohn commented on pilot/pkg/xds/ads.go:
```diff
@@ -413,41 +380,15 @@ func (s *DiscoveryServer) handleCds(con *Connection, discReq *discovery.Discover
-			con.mu.Unlock()
-		}
+	// TODO(ramaraochavali): This special handling is not needed for EDS. But tests are failing without this.
+	// Need to investigate if there is a bug in adsc.
```
There is - it doesn't return resource names on ACK

howardjohn commented on pilot/pkg/xds/ads.go:
```diff
@@ -502,6 +410,65 @@ func (s *DiscoveryServer) handleRds(con *Connection, discReq *discovery.Discover
+
+	previousInfo := con.Active[request.TypeUrl]
+
+	// This is not a first request, we should see if it is a ACK/NACK.
```
nit: already checked NACK

howardjohn commented on pilot/pkg/xds/ads.go:
```diff
@@ -502,6 +410,65 @@ func (s *DiscoveryServer) handleRds(con *Connection, discReq *discovery.Discover
+			return false
+		}
+
+		// If it comes here, that means nonce and version match. This an ACK. We should
```
nit: we don't actually check version, so the comment is not quite correct

howardjohn commented on pilot/pkg/xds/ads.go:
```diff
@@ -502,6 +410,65 @@ func (s *DiscoveryServer) handleRds(con *Connection, discReq *discovery.Discover
+		// If it comes here, that means nonce and version match. This an ACK. We should
+		// record the ack details and respond if there is a change in resource names.
+		con.mu.Lock()
+		if con.Active[request.TypeUrl] == nil {
```
should key be short type? otherwise a mixed v2 and v3 will break things?

howardjohn commented on pilot/pkg/xds/debug.go:
```diff
@@ -161,14 +162,14 @@ func (s *DiscoveryServer) Syncz(w http.ResponseWriter, _ *http.Request) {
-				RouteAcked:    con.RouteNonceAcked,
-				EndpointSent:  con.EndpointNonceSent,
-				EndpointAcked: con.EndpointNonceAcked,
+				ClusterSent:   con.NonceSent(v3.ClusterType),
```
this will break v2

howardjohn replied:
See comment above about using short type

istio-testing commented:
@ramaraochavali: PR needs rebase.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

istio-testing commented:
@ramaraochavali: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/unit-tests_istio/15947) | `/test unit-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@ramaraochavali: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/unit-tests_istio/15947) | `/test unit-tests_istio`
integ-telemetry-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-telemetry-k8s-tests_istio/15380) | `/test integ-telemetry-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@ramaraochavali: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/unit-tests_istio/15947) | `/test unit-tests_istio`
integ-telemetry-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-telemetry-k8s-tests_istio/15380) | `/test integ-telemetry-k8s-tests_istio`
integ-galley-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-galley-k8s-tests_istio/15366) | `/test integ-galley-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@ramaraochavali: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/unit-tests_istio/15947) | `/test unit-tests_istio`
integ-telemetry-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-telemetry-k8s-tests_istio/15380) | `/test integ-telemetry-k8s-tests_istio`
integ-galley-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-galley-k8s-tests_istio/15366) | `/test integ-galley-k8s-tests_istio`
integ-operator-controller-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-operator-controller-tests_istio/3817) | `/test integ-operator-controller-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@ramaraochavali: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/unit-tests_istio/15947) | `/test unit-tests_istio`
integ-telemetry-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-telemetry-k8s-tests_istio/15380) | `/test integ-telemetry-k8s-tests_istio`
integ-galley-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-galley-k8s-tests_istio/15366) | `/test integ-galley-k8s-tests_istio`
integ-operator-controller-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-operator-controller-tests_istio/3817) | `/test integ-operator-controller-tests_istio`
release-test_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/release-test_istio/15254) | `/test release-test_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@ramaraochavali: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/unit-tests_istio/15947) | `/test unit-tests_istio`
integ-telemetry-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-telemetry-k8s-tests_istio/15380) | `/test integ-telemetry-k8s-tests_istio`
integ-galley-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-galley-k8s-tests_istio/15366) | `/test integ-galley-k8s-tests_istio`
integ-operator-controller-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-operator-controller-tests_istio/3817) | `/test integ-operator-controller-tests_istio`
release-test_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/release-test_istio/15254) | `/test release-test_istio`
integ-multicluster-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-multicluster-k8s-tests_istio/5965) | `/test integ-multicluster-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@ramaraochavali: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/unit-tests_istio/15947) | `/test unit-tests_istio`
integ-telemetry-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-telemetry-k8s-tests_istio/15380) | `/test integ-telemetry-k8s-tests_istio`
integ-galley-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-galley-k8s-tests_istio/15366) | `/test integ-galley-k8s-tests_istio`
integ-operator-controller-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-operator-controller-tests_istio/3817) | `/test integ-operator-controller-tests_istio`
release-test_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/release-test_istio/15254) | `/test release-test_istio`
integ-multicluster-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-multicluster-k8s-tests_istio/5965) | `/test integ-multicluster-k8s-tests_istio`
integ-distroless-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-distroless-k8s-tests_istio/8624) | `/test integ-distroless-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@ramaraochavali: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/unit-tests_istio/15947) | `/test unit-tests_istio`
integ-telemetry-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-telemetry-k8s-tests_istio/15380) | `/test integ-telemetry-k8s-tests_istio`
integ-galley-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-galley-k8s-tests_istio/15366) | `/test integ-galley-k8s-tests_istio`
integ-operator-controller-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-operator-controller-tests_istio/3817) | `/test integ-operator-controller-tests_istio`
release-test_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/release-test_istio/15254) | `/test release-test_istio`
integ-multicluster-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-multicluster-k8s-tests_istio/5965) | `/test integ-multicluster-k8s-tests_istio`
integ-distroless-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-distroless-k8s-tests_istio/8624) | `/test integ-distroless-k8s-tests_istio`
integ-ipv6-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-ipv6-k8s-tests_istio/1636) | `/test integ-ipv6-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@ramaraochavali: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/unit-tests_istio/15947) | `/test unit-tests_istio`
integ-telemetry-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-telemetry-k8s-tests_istio/15380) | `/test integ-telemetry-k8s-tests_istio`
integ-galley-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-galley-k8s-tests_istio/15366) | `/test integ-galley-k8s-tests_istio`
integ-operator-controller-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-operator-controller-tests_istio/3817) | `/test integ-operator-controller-tests_istio`
release-test_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/release-test_istio/15254) | `/test release-test_istio`
integ-multicluster-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-multicluster-k8s-tests_istio/5965) | `/test integ-multicluster-k8s-tests_istio`
integ-distroless-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-distroless-k8s-tests_istio/8624) | `/test integ-distroless-k8s-tests_istio`
integ-ipv6-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-ipv6-k8s-tests_istio/1636) | `/test integ-ipv6-k8s-tests_istio`
lint_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/lint_istio/15383) | `/test lint_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@ramaraochavali: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/unit-tests_istio/15947) | `/test unit-tests_istio`
integ-telemetry-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-telemetry-k8s-tests_istio/15380) | `/test integ-telemetry-k8s-tests_istio`
integ-galley-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-galley-k8s-tests_istio/15366) | `/test integ-galley-k8s-tests_istio`
integ-operator-controller-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-operator-controller-tests_istio/3817) | `/test integ-operator-controller-tests_istio`
release-test_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/release-test_istio/15254) | `/test release-test_istio`
integ-multicluster-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-multicluster-k8s-tests_istio/5965) | `/test integ-multicluster-k8s-tests_istio`
integ-distroless-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-distroless-k8s-tests_istio/8624) | `/test integ-distroless-k8s-tests_istio`
integ-ipv6-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-ipv6-k8s-tests_istio/1636) | `/test integ-ipv6-k8s-tests_istio`
lint_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/lint_istio/15383) | `/test lint_istio`
gencheck_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/gencheck_istio/13974) | `/test gencheck_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@ramaraochavali: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/unit-tests_istio/15947) | `/test unit-tests_istio`
integ-telemetry-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-telemetry-k8s-tests_istio/15380) | `/test integ-telemetry-k8s-tests_istio`
integ-galley-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-galley-k8s-tests_istio/15366) | `/test integ-galley-k8s-tests_istio`
integ-operator-controller-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-operator-controller-tests_istio/3817) | `/test integ-operator-controller-tests_istio`
release-test_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/release-test_istio/15254) | `/test release-test_istio`
integ-multicluster-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-multicluster-k8s-tests_istio/5965) | `/test integ-multicluster-k8s-tests_istio`
integ-distroless-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-distroless-k8s-tests_istio/8624) | `/test integ-distroless-k8s-tests_istio`
integ-ipv6-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-ipv6-k8s-tests_istio/1636) | `/test integ-ipv6-k8s-tests_istio`
lint_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/lint_istio/15383) | `/test lint_istio`
gencheck_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/gencheck_istio/13974) | `/test gencheck_istio`
integ-security-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-security-k8s-tests_istio/15644) | `/test integ-security-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@ramaraochavali: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/unit-tests_istio/15947) | `/test unit-tests_istio`
integ-telemetry-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-telemetry-k8s-tests_istio/15380) | `/test integ-telemetry-k8s-tests_istio`
integ-galley-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-galley-k8s-tests_istio/15366) | `/test integ-galley-k8s-tests_istio`
integ-operator-controller-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-operator-controller-tests_istio/3817) | `/test integ-operator-controller-tests_istio`
release-test_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/release-test_istio/15254) | `/test release-test_istio`
integ-multicluster-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-multicluster-k8s-tests_istio/5965) | `/test integ-multicluster-k8s-tests_istio`
integ-distroless-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-distroless-k8s-tests_istio/8624) | `/test integ-distroless-k8s-tests_istio`
integ-ipv6-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-ipv6-k8s-tests_istio/1636) | `/test integ-ipv6-k8s-tests_istio`
lint_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/lint_istio/15383) | `/test lint_istio`
gencheck_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/gencheck_istio/13974) | `/test gencheck_istio`
integ-security-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-security-k8s-tests_istio/15644) | `/test integ-security-k8s-tests_istio`
integ-pilot-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-pilot-k8s-tests_istio/15504) | `/test integ-pilot-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@ramaraochavali: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/unit-tests_istio/15947) | `/test unit-tests_istio`
integ-telemetry-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-telemetry-k8s-tests_istio/15380) | `/test integ-telemetry-k8s-tests_istio`
integ-galley-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-galley-k8s-tests_istio/15366) | `/test integ-galley-k8s-tests_istio`
integ-operator-controller-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-operator-controller-tests_istio/3817) | `/test integ-operator-controller-tests_istio`
release-test_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/release-test_istio/15254) | `/test release-test_istio`
integ-multicluster-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-multicluster-k8s-tests_istio/5965) | `/test integ-multicluster-k8s-tests_istio`
integ-distroless-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-distroless-k8s-tests_istio/8624) | `/test integ-distroless-k8s-tests_istio`
integ-ipv6-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-ipv6-k8s-tests_istio/1636) | `/test integ-ipv6-k8s-tests_istio`
lint_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/lint_istio/15383) | `/test lint_istio`
gencheck_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/gencheck_istio/13974) | `/test gencheck_istio`
integ-security-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-security-k8s-tests_istio/15644) | `/test integ-security-k8s-tests_istio`
integ-pilot-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-pilot-k8s-tests_istio/15504) | `/test integ-pilot-k8s-tests_istio`
integ-mixer-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-mixer-k8s-tests_istio/13848) | `/test integ-mixer-k8s-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

howardjohn commented:
If you use the existing active I am pretty sure there is a race condition
where OnConnect will access the Active map in a different goroutine - if
you look at my pr about the typeurl there is some commentary. My solution
was not great...

On Mon, Jul 6, 2020, 8:44 PM Istio Automation <notifications@github.com>
wrote:

> @ramaraochavali <https://github.com/ramaraochavali>: The following tests
> *failed*, say /retest to rerun all failed tests:
> Test name Commit Details Rerun command
> unit-tests_istio ec73d42
> <https://github.com/istio/istio/commit/ec73d426db750cc9a13e648f35683e8159185c28>
> link
> <https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/unit-tests_istio/15947> /test
> unit-tests_istio
> integ-telemetry-k8s-tests_istio ec73d42
> <https://github.com/istio/istio/commit/ec73d426db750cc9a13e648f35683e8159185c28>
> link
> <https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-telemetry-k8s-tests_istio/15380> /test
> integ-telemetry-k8s-tests_istio
> integ-galley-k8s-tests_istio ec73d42
> <https://github.com/istio/istio/commit/ec73d426db750cc9a13e648f35683e8159185c28>
> link
> <https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-galley-k8s-tests_istio/15366> /test
> integ-galley-k8s-tests_istio
> integ-operator-controller-tests_istio ec73d42
> <https://github.com/istio/istio/commit/ec73d426db750cc9a13e648f35683e8159185c28>
> link
> <https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-operator-controller-tests_istio/3817> /test
> integ-operator-controller-tests_istio
> release-test_istio ec73d42
> <https://github.com/istio/istio/commit/ec73d426db750cc9a13e648f35683e8159185c28>
> link
> <https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/release-test_istio/15254> /test
> release-test_istio
> integ-multicluster-k8s-tests_istio ec73d42
> <https://github.com/istio/istio/commit/ec73d426db750cc9a13e648f35683e8159185c28>
> link
> <https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-multicluster-k8s-tests_istio/5965> /test
> integ-multicluster-k8s-tests_istio
> integ-distroless-k8s-tests_istio ec73d42
> <https://github.com/istio/istio/commit/ec73d426db750cc9a13e648f35683e8159185c28>
> link
> <https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-distroless-k8s-tests_istio/8624> /test
> integ-distroless-k8s-tests_istio
> integ-ipv6-k8s-tests_istio ec73d42
> <https://github.com/istio/istio/commit/ec73d426db750cc9a13e648f35683e8159185c28>
> link
> <https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-ipv6-k8s-tests_istio/1636> /test
> integ-ipv6-k8s-tests_istio
> lint_istio ec73d42
> <https://github.com/istio/istio/commit/ec73d426db750cc9a13e648f35683e8159185c28>
> link
> <https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/lint_istio/15383> /test
> lint_istio
> gencheck_istio ec73d42
> <https://github.com/istio/istio/commit/ec73d426db750cc9a13e648f35683e8159185c28>
> link
> <https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/gencheck_istio/13974> /test
> gencheck_istio
> integ-security-k8s-tests_istio ec73d42
> <https://github.com/istio/istio/commit/ec73d426db750cc9a13e648f35683e8159185c28>
> link
> <https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-security-k8s-tests_istio/15644> /test
> integ-security-k8s-tests_istio
> integ-pilot-k8s-tests_istio ec73d42
> <https://github.com/istio/istio/commit/ec73d426db750cc9a13e648f35683e8159185c28>
> link
> <https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-pilot-k8s-tests_istio/15504> /test
> integ-pilot-k8s-tests_istio
> integ-mixer-k8s-tests_istio ec73d42
> <https://github.com/istio/istio/commit/ec73d426db750cc9a13e648f35683e8159185c28>
> link
> <https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-mixer-k8s-tests_istio/13848> /test
> integ-mixer-k8s-tests_istio
>
> Instructions for interacting with me using PR comments are available here
> <https://git.k8s.io/community/contributors/guide/pull-requests.md>. If
> you have questions or suggestions related to my behavior, please file an
> issue against the kubernetes/test-infra
> <https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
> repository. I understand the commands that are listed here
> <https://go.k8s.io/bot-commands>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/istio/istio/pull/25192#issuecomment-654581017>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAEYGXK2ZMCNEQY76REEEJTR2KKZLANCNFSM4OPWLXIQ>
> .
>

istio-testing commented:
@ramaraochavali: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/unit-tests_istio/15947) | `/test unit-tests_istio`
integ-telemetry-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-telemetry-k8s-tests_istio/15380) | `/test integ-telemetry-k8s-tests_istio`
integ-galley-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-galley-k8s-tests_istio/15366) | `/test integ-galley-k8s-tests_istio`
integ-operator-controller-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-operator-controller-tests_istio/3817) | `/test integ-operator-controller-tests_istio`
release-test_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/release-test_istio/15254) | `/test release-test_istio`
integ-multicluster-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-multicluster-k8s-tests_istio/5965) | `/test integ-multicluster-k8s-tests_istio`
integ-distroless-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-distroless-k8s-tests_istio/8624) | `/test integ-distroless-k8s-tests_istio`
integ-ipv6-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-ipv6-k8s-tests_istio/1636) | `/test integ-ipv6-k8s-tests_istio`
integ-security-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-security-k8s-tests_istio/15644) | `/test integ-security-k8s-tests_istio`
integ-pilot-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-pilot-k8s-tests_istio/15504) | `/test integ-pilot-k8s-tests_istio`
integ-mixer-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-mixer-k8s-tests_istio/13848) | `/test integ-mixer-k8s-tests_istio`
gencheck_istio | 730ce81d0eacb9f5ec7ad47c6e1fc26344802696 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/gencheck_istio/13975) | `/test gencheck_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@ramaraochavali: The following tests **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integ-telemetry-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-telemetry-k8s-tests_istio/15380) | `/test integ-telemetry-k8s-tests_istio`
integ-galley-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-galley-k8s-tests_istio/15366) | `/test integ-galley-k8s-tests_istio`
integ-operator-controller-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-operator-controller-tests_istio/3817) | `/test integ-operator-controller-tests_istio`
release-test_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/release-test_istio/15254) | `/test release-test_istio`
integ-multicluster-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-multicluster-k8s-tests_istio/5965) | `/test integ-multicluster-k8s-tests_istio`
integ-distroless-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-distroless-k8s-tests_istio/8624) | `/test integ-distroless-k8s-tests_istio`
integ-ipv6-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-ipv6-k8s-tests_istio/1636) | `/test integ-ipv6-k8s-tests_istio`
integ-security-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-security-k8s-tests_istio/15644) | `/test integ-security-k8s-tests_istio`
integ-pilot-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-pilot-k8s-tests_istio/15504) | `/test integ-pilot-k8s-tests_istio`
integ-mixer-k8s-tests_istio | ec73d426db750cc9a13e648f35683e8159185c28 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/integ-mixer-k8s-tests_istio/13848) | `/test integ-mixer-k8s-tests_istio`
gencheck_istio | 730ce81d0eacb9f5ec7ad47c6e1fc26344802696 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/gencheck_istio/13975) | `/test gencheck_istio`
unit-tests_istio | 730ce81d0eacb9f5ec7ad47c6e1fc26344802696 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/25192/unit-tests_istio/15948) | `/test unit-tests_istio`

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

ramaraochavali commented:
Thanks for pointing that out. I think having separate watched resources for experimental makes sense for now instead of complicating it. I have followed the same pattern

howardjohn commented:
Ok sgtm. Haven't looked at the code yet - I assume that means my pr for the
type url is compatible with this? Or perhaps it's obsolete. Either way,
I'll hopefully approve this tomorrow then can merge my on top? Does that
make sense?



On Mon, Jul 6, 2020, 10:10 PM Rama Chavali <notifications@github.com> wrote:

> Thanks for pointing that out. I think having separate watched resources
> for experimental makes sense for now instead of complicating it. I have
> followed the same pattern
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/istio/istio/pull/25192#issuecomment-654604366>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAEYGXOM6DCWBKWKOCBYYTDR2KU3TANCNFSM4OPWLXIQ>
> .
>

ramaraochavali commented:
Yes. That makes sense. PTAL in detail whenever you get time. No rush
